### PR TITLE
Makes VolumeNameGenerator public

### PR DESCRIPTION
**Pull request by los93sol**
Makes VolumeNameGenerator public #6490 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6531)

### CODE REVIEW

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs [hunk 1]
```diff
@@ -5,16 +5,26 @@
...
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="suffix">The suffix to append to the volume name.</param>
+    /// <returns>The volume name.</returns>
+    /// <exception cref="ArgumentException"></exception>
     public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suffix) where T : IResource
     {
         if (!HasOnlyValidChars(suffix))
         {
             throw new ArgumentException($"The suffix '{suffix}' contains invalid characters. Only [a-zA-Z0-9_.-] are allowed.", nameof(suffix));
```
Comment (danmoseley) [inline src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:25]:
```suggestion
            throw new ArgumentException($"The suffix '{suffix}' contains invalid characters. It must match [a-zA-Z0-9][a-zA-Z0-9_.-]*.", nameof(suffix));
```
perhaps?

Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs]: I like that, thanks!

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs [hunk 2]
```diff
@@ -24,6 +34,11 @@ public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suf
         return $"{safeApplicationName}-{applicationHash}-{resourceName}-{suffix}";
     }
 
+    /// <summary>
+    /// Sanitizes the application name to be used in the volume name.
+    /// </summary>
+    /// <param name="name">The application name to be sanitized.</param>
+    /// <returns>The sanitized application name.</returns>
```
Comment (danmoseley) [inline src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:41]: Sanitize seems it does not need to be public.

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs [hunk 3]
```diff
@@ -5,16 +5,26 @@
...
 {
+    /// <summary>
+    /// Creates a volume name with the form <c>{applicationName}-{sha256 of apphost path}-{resourceName}-{suffix}</c>, e.g. <c>myapplication-a345f2451-postgres-data</c>.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="suffix">The suffix to append to the volume name.</param>
+    /// <returns>The volume name.</returns>
+    /// <exception cref="ArgumentException"></exception>
     public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suffix) where T : IResource
```
Comment (danmoseley) [inline src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:21]: Probably want to think about naming here.. duplicative ? `VolumeNameGenerator.Generate` ?

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs (hunk 2 above)
Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:41]: Agree, there’s some existing things referencing it, I’ll go through those and see what needs to happen to make it private

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs (hunk 3 above)
Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:21]: I have no preference on this, simply changed the visibility.  I agree though, Generate seems more consistent and it’s an opportunity to polish it a bit.  I’ll update it

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs (hunk 2 above)
Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:41]: I updated to change the visibility of Sanitize to internal

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs (hunk 3 above)
Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs]: I have updated to change the name from CreateVolumeName to Generate as suggested.

Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs]: @danmoseley Are you good with the changes made?

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs (hunk 1 above)
Comment (danmoseley) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:25]: I don't see this change for some reason.

**Comment by davidfowl** [comment]
cc @DamianEdwards as he wrote this code

File: src/Aspire.Hosting/Utils/VolumeNameGenerator.cs [hunk 4]
```diff
@@ -24,7 +34,12 @@ public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suf
         return $"{safeApplicationName}-{applicationHash}-{resourceName}-{suffix}";
     }
 
-    public static string Sanitize(string name)
+    /// <summary>
+    /// Sanitizes the application name to be used in the volume name.
+    /// </summary>
+    /// <param name="name">The application name to be sanitized.</param>
+    /// <returns>The sanitized application name.</returns>
+    internal static string Sanitize(string name)
```
[the remark this answers is not in the document]
Comment (los93sol) [reply src/Aspire.Hosting/Utils/VolumeNameGenerator.cs:37]: I removed the comment

**Review by danmoseley — APPROVED** [review]
thanks

**Comment by danmoseley** [comment]
@los93sol if you put text like `Fixes #6490` in your top post in the PR, merging the PR closes the issue.

Thanks for the contribution, we'd welcome more.

**Pull request merged**